### PR TITLE
refactor(core): add CoreAPI class

### PR DESCRIPTION
**Pull request by ender503**
To enhance the code readability, modify the composeAPI to return an
CoreAPI instance instead of an object.

### CODE REVIEW

File: packages/core/src/CoreAPI.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  private registry: IdenityRegistry;
+  private documentLoader: DocumentLoader;
+
+  constructor(settings: Partial<Settings> = {}) {
+    this.registry = new IdenityRegistry({ provider: settings.provider });
+    this.documentLoader = new DocumentLoader(this.registry);
+  }
+
+  /**
+   * Publish the DID document to the Tangle MAM channel with specific network.
```
Comment (jserv) [inline packages/core/src/CoreAPI.ts]: Change from "Tangle MAM channel with specific network" to "MAM channel with specific Tangle network". Please be aware of the fact of second layer for MAM channel.

File: packages/core/src/CoreAPI.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import { IdenityRegistry } from '@tangleid/did';
+import { signRsaSignature, verifyRsaSignature } from '@tangleid/jsonld';
+import { DocumentLoader } from './jsonld/documentLoader';
+
+import { IriProvider, Seed, Did, PublicKeyMeta, PrivateKeyPem, PublicKeyPem } from '../../types';
```
Comment (jserv) [inline packages/core/src/CoreAPI.ts:5]: Again. Send another pull request for revising `IriProvider`. Something like `TangleEndpoint` might be better.

File: packages/core/src/CoreAPI.ts (hunk 1 above)
Comment (ender503) [reply packages/core/src/CoreAPI.ts]: Fixed.

File: packages/core/src/CoreAPI.ts (hunk 2 above)
Comment (ender503) [reply packages/core/src/CoreAPI.ts:5]: Revised at #172 .
